pynance/textimporter.py

```diff
@@ -0,0 +1,205 @@
+import csv
+import datetime
+
+import pandas as pd
+import numpy as np
+
+
+class CsvImporter():
+    """
+    Importer class to read csv files to well defined pandas.DataFrames
+    """
+    def __init__(self):
+        pass
+
+    def read(self, filepath_or_buffer, cfd):
+        """
+        Reads the text in a csv file or buffer and converts it into a
+        DataFrame as specified by a CsvFileDescription
+
+        Parameters
+        ----------
+        filepath_or_buffer : str, pathlib.Path, py._path.local.LocalPath or
+            any object with a read() method.
+            Passed to pandas.read_csv
+        cfd : CsvFileDescription, a description of how the CSV file is to be
+            read and transformed
+        """
+        columns_to_read = cfd.column_map.values()
+
+        # read the dataframe as it is, with only strings
+        # formatting is done later
+        df_as_is = pd.read_csv(filepath_or_buffer=filepath_or_buffer,
+                               dialect=cfd.csv_dialect,
+                               skiprows=cfd.skiprows,
+                               encoding=cfd.encoding,
+                               usecols=columns_to_read,
+                               dtype=str)
+
+        # construct a new DataFrame that matches the definitions
+        new_df = pd.DataFrame()
+
+        # iterate over all the columns that we want to have in the end
+        for new_col_name, new_type in COLUMNS.items():
+
+            if new_col_name in cfd.column_map.keys():
+                old_col_name = cfd.column_map[new_col_name]
+
+                if old_col_name not in df_as_is.columns.values:
+                    raise UnsupportedCsvFormat("Column %s was not found in \
+                                            file" % old_col_name)
+
+                # get the formatter for the required type
+                formatter = cfd.formatters[new_type]
```

Comment:
What if key `new_type` does not exist?

Reply:
First I wrote a try..catch here, but in fact this would not be a problem of the user, but of the implementation - so it should be checked during construction of description or in tests, I think

Reply:
the `assert` statement in the constructor of `CsvFileDescription` should do the job

Reply:
Yes, the `assert` is a nice fix imo.